account-kit/react/src/components/auth/card/loading/oauth.tsx

```diff
@@ -0,0 +1,65 @@
+import { useEffect } from "react";
+import { useSignerStatus } from "../../../../hooks/useSignerStatus.js";
+import { useAuthContext, type AuthStep } from "../../context.js";
+import { ContinueWithOAuth } from "../../../../icons/oauth.js";
+import { ConnectionError } from "../error/connection-error.js";
+
+interface CompletingOAuthProps {
+  authStep: Extract<AuthStep, { type: "oauth_completing" }>;
+}
+
+export const CompletingOAuth = ({ authStep }: CompletingOAuthProps) => {
+  const { isConnected } = useSignerStatus();
+  const { setAuthStep } = useAuthContext();
+
+  useEffect(() => {
+    if (isConnected) {
+      setAuthStep({ type: "complete" });
+    }
+  }, [isConnected, setAuthStep]);
+
+  if (authStep.error) {
+    return (
+      <ConnectionError
+        connectionType="oauth"
+        handleUseAnotherMethod={() => setAuthStep({ type: "complete" })}
+      />
+    );
+  }
+
+  const capitalize = (str: string) => {
+    return str
+      .split(" ")
+      .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
+      .join(" ");
+  };
```

Comment:
nit: pull this out of the component since you don't need this instantiated each render.